packages/payment-experience-api/src/api/createRefundController.ts

```diff
@@ -0,0 +1,175 @@
+import {
+  AbstractController,
+  Data,
+  ExperienceError,
+  RequestValidationError,
+  UnexpectedError,
+  ValidatedRequest,
+} from '@verkkokauppa/core'
+import * as yup from 'yup'
+import type { Response } from 'express'
+import { validateApiKey } from '@verkkokauppa/configuration-backend'
+import {
+  confirmRefundAdmin,
+  createRefund,
+  getOrderAdmin,
+  getRefundsByOrderAdmin,
+  OrderItem,
+  Refund,
+  RefundItem,
+} from '@verkkokauppa/order-backend'
+import {
+  createRefundPaymentFromRefund,
+  getPaidPaymentAdmin,
+  paidPaymentExists,
+  RefundGateway,
+} from '@verkkokauppa/payment-backend'
+import { calculateTotalsFromItems } from '../../../order-experience-api/dist/lib/totals'
+
+const requestSchema = yup.object().shape({
+  params: yup.object().shape({
+    orderId: yup.string().required(),
+  }),
+  headers: yup.object().shape({
+    'api-key': yup.string().required(),
+    namespace: yup.string().required(),
+  }),
+})
+
+export class CreateRefundController extends AbstractController<
```

Comment:
Should this be renamed to InstatRefundController etc?

Reply:
It would be more fitting name considering how refunds are handled - I'll rename it.